### PR TITLE
add last few weeks of whiteboarding

### PR DESCRIPTION
**Pull request by ggorlen**

### CODE REVIEW

File: 2018-12-12/concurrent_sum.py [hunk 1]
```diff
@@ -0,0 +1,144 @@
+"""
+12/12/2018: Problem 4 solution
+
+Given 100 computers that each gives you a number when you make a function call with the computer's ID, sum up the numbers.
```
Comment (elimisteve) [inline 2018-12-12/concurrent_sum.py:4]: I recommend making these lines shorter :-)

File: 2018-12-12/concurrent_sum.py [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+    would not make the CPU do heavy work.
+    (The Global Interpreter Lock in Python
+    prevents code from running in parallel,
+    so using threads does not speed up CPU-intensive work,
+    but it does speed up I/O work since it is not done by the CPU.)
+
+    Author: Eugene Ma
+    """
+
+    import random
```
Comment (elimisteve) [inline 2018-12-12/concurrent_sum.py:27]: Import at the top of the file

File: 2018-12-12/concurrent_sum.py [hunk 3]
```diff
@@ -0,0 +1,144 @@
...
+    for compId in range(NUM_COMPS):
+        success = False
+        attempt = 0
+        while not success:
+            attempt += 1
+            try:
+                total += getNumFromCompId(compId)
+                success = True
+            except IOError as err:
+                print("Attempt {0}: {1}".format(attempt, err))
```
Comment (elimisteve) [inline 2018-12-12/concurrent_sum.py:69]: Could have done

```python
print("Attempt {}: {}".format(attempt, err))
```

File: 2018-12-12/force_order_in_concurrency.py [hunk 4]
```diff
@@ -0,0 +1,88 @@
+"""
+12/12/2018: Problem 5
+
+CTCI concurrency problem: Given a class that has methods `method1()`, `method2()`, and `method3()`, each of these methods will be run on a separate thread. Devise a mechanism where `method2()` only starts after `method1()` is finished and `method3()` starts only after `method2()` is finished.
```
Comment (elimisteve) [inline 2018-12-12/force_order_in_concurrency.py:4]: Wrap

**Comment by elimisteve** [comment]
@ggorlen @eugene-ma Merging now, but check out my comments for recommended styling tweaks :-)

File: 2018-12-12/concurrent_sum.py (hunk 3 above)
Comment (ggorlen) [reply 2018-12-12/concurrent_sum.py:69]: Or `print("Attempt %s: %s" % (attempt, err))`

File: 2018-12-12/concurrent_sum.py (hunk 1 above)
Comment (ggorlen) [reply 2018-12-12/concurrent_sum.py:4]: Good call.

**Comment by ggorlen** [comment]
Thanks--great points all around! Fixed wraps (my fault from a copy and paste). I'm leaving code pulls from other participants as-is but did point to http://google.github.io/styleguide/pyguide.html when I accepted the last PR.

**Pull request merged**